### DISCUSSION TITLE
K3s customer adoption and ecosystem support

### DISCUSSION THREAD

**Discussion by friggitello6** [k3s-io/k3s#6533] [Q&A] [answered]
This is not a feature request per se, more like some kind of white paper on current K3s status on customer adoption and ecosystem support.  We are working on an edge solution and need to decide whether it should support K3s.  If yes, we would like to go one step further and explore whether K3s can be a replacement for K8s or not so that we don't need to support both.  Here is a list of things that can help us make educated decision:
1. List of open-source projects that support K3s.
2. List of critical open-source projects that don't support K3s, but plan to support in their roadmap.
3. List of critical open-source projects that don't support K3s, and there is no plan to support it in their roadmap.
4. List of big ISVs that support K3s.
5. List of big ISVs that don't support K3s, but plan to support in their roadmap.
6. List of big ISVs that don't support K3s, and there is no plan to support it in their roadmap.
7. What are the main reasons that ISVs don't support K3s?  Is it because of the amount of certification/testing?
8. Is there any potential compatibility issue that a workload can run on K8s, but not K3s.

**Comment by brandond** [comment, marked as answer]
K3 is Kubernetes (K8s). Anything that works on any other [CNCF-certified ](https://www.cncf.io/certification/software-conformance/)Kubernetes distro should work on K3s. If something does not, please feel free to open an issue. We don't actively maintain lists of projects or companies that "support" K3s because that is not something that makes any sense for us to spend time doing.

If you're looking for information on paid support, this open-source project is not the correct place to inquire about that. Most of the team works for SUSE, who does offer paid support. You can find the contact request form at https://www.suse.com/contact/

**Reply by friggitello6** [reply]
Understood.  I feel such requested information will help adoption of K3s and should be of best interest to this open-source community.

**Reply by cwayne18** [reply]
We have an adopters.md file in the repo, that's really as far as we feel we should go on this.

**Reply by friggitello6** [reply]
Thank you @cwayne18  @brandond.  Is there a way I can schedule a call with one of you?

**Reply by brandond** [reply]
To what end? I believe your question has been answered here, to the extent we are willing to answer it. If you want additional offline discussion, you're welcome to join the `#k3s` channel on the Rancher Users slack: https://slack.rancher.io/. 

Most of what you're asking about is what I would traditionally view as pre-sales due diligence, not an engineering question, for which I would direct you back to the contact URL linked above.

**Reply by friggitello6** [reply]
As always, thanks for the quick response.  I checked the adopters.md file and the list is very tiny.  It looks like it is true that K3s lacks ecosystem support, and it is not something any big company like Microsoft should bet on at this point, right?

**Reply by brandond** [reply]
I'm not sure why you are asking about ecosystem support. A K3s cluster is Kubernetes, same as any other conformant cluster such as those provisioned by kubeadm, minikube, kubespray, microk8s, EKS, GKE, or any other distro listed at https://www.cncf.io/certification/software-conformance/#logos. No one in the "ecosystem" needs to specifically support K3s as long as their product works on Kubernetes. Is there something about this that is confusing?

Users are not required to add themselves to the adopters file, so it essentially only lists those that want to advertise their use of K3s. It's funny that you would mention Microsoft, as their AKS Lite product (currently in public preview) is actually built around K3s.

https://techcommunity.microsoft.com/t5/internet-of-things-blog/taking-azure-arc-and-kubernetes-to-the-edge/ba-p/3650599

**Reply by friggitello6** [reply]
Yes, Microsoft have AKS Lite that supports both K8s and K3s.  We are working on another very important edge initiative and there is huge debate internally on whether we should support K3s at all or present K3s as the only option or support both K8s and K3s.  And there is strong push back on supporting K3s because of concerns around ecosystem support and customer adoption.  Information to the above questions will help us make educated decision.

**Reply by brandond** [reply]
I think we're at a bit of an impasse. We have made it clear that the engineering team that maintains K3s does not currently collect, nor do we have plans to being collecting, the information you're asking for. 

What organization do you represent? Your github profile lacks any identifying information whatsoever, and you are being unusually coy about the nature of your project. Normally an inquiry of this importance would be handled via direct contact with someone from SUSE/Rancher, not by an anonymous commenter asking oblique questions on Github. If you'd like to continue this discussion, I again point you at https://www.suse.com/contact/ and suggest that you reach out to someone who may be able to provide competitive strategy information.

**Reply by friggitello6** [reply]
Thanks.  I will use https://www.suse.com/contact/ then.

**Reply by brandond** [reply]
This wouldn't by any chance be related to a class assignment or other coursework, would it?

**Reply by friggitello6** [reply]
No, I have no idea what you are thinking about.  Anyway, I shared my contact through the above link and added this discussion in my request.  Not sure if any SUSE's people will reach out to you to share the context.

Thank You!
A SUSE representative will be in touch shortly. Thank you for your interest in SUSE.
